oshi-core/src/main/java/oshi/software/os/OSProcess.java

```diff
@@ -674,4 +723,48 @@ public int getBitness() {
     public void setBitness(int bitness) {
         this.bitness = bitness;
     }
+
+    private OperatingSystem getCurrentOperatingSystem() {
+        switch (SystemInfo.getCurrentPlatformEnum()) {
+            case WINDOWS:
+                return new WindowsOperatingSystem();
+            case LINUX:
+                return new LinuxOperatingSystem();
+            case MACOSX:
+                return new MacOperatingSystem();
+            case SOLARIS:
+                return new SolarisOperatingSystem();
+            case FREEBSD:
+                return new FreeBsdOperatingSystem();
+            default:
+                return null;
```

Comment:
This should never happen unless we add a new OS in the platform enum and don't implement it here, so it's probably safe to throw an exception here, and then you don't have to handle the null return value.